frontend/src/app/main/data/workspace/interactions.cljs

```diff
@@ -66,22 +58,13 @@
   (ptk/reify ::remove-flow
     ptk/WatchEvent
     (watch [it state _]
-      (let [page-id (:current-page-id state)
-            flows   (get-in state [:workspace-data
-                                   :pages-index
-                                   page-id
-                                   :options
-                                   :flows] [])]
+      (let [page    (wsh/lookup-page state)
+            flows   (get-in page [:options :flows] [])]
         (rx/of (dch/commit-changes
-                {:redo-changes [{:type :set-option
-                                 :page-id page-id
-                                 :option :flows
-                                 :value (csp/remove-flow flows flow-id)}]
-                 :undo-changes [{:type :set-option
-                                 :page-id page-id
-                                 :option :flows
-                                 :value flows}]
-                 :origin it}))))))
+                 (-> (pcb/empty-changes it)
+                     (pcb/with-page page)
+                     (pcb/set-page-option :flows
+                                          (csp/remove-flow flows flow-id)))))))))
 
```

Comment:
Look os as-is, but I think this can be done with different apprach. I mean, why we don't have a specific operations for removing flows?

Something like this:
```
(-> (pcb/empty-changes it)
    (pcb/with-page page)
    (pcb/update-page-options csp/remove-flow flow-id))
```

The `update-page-options` will work in the same way as the clojure `update`, will receive the accesor function, and the additional arguments, and makes it more ergonomic, and you will not need to retrieve the flow previously from page.

Reply:
Ok. I'll leave set-option for simple options with a key and a static, and write new update-page-options for more complex cases

Reply:
I think update is pretty generic, and can be used together with assoc for setting an option.
```
(pcb/update-page-options assoc :test 1)
```